src/main/java/com/revature/services/EvalLogicImpl.java

```diff
@@ -16,8 +22,93 @@
 	@Autowired
 	private EvalRepository dao;
 	
+	@Autowired
+	private EvalCommentRepository commentDao;
+	
+	@PersistenceContext
+    private EntityManager entityManager;
+	
 
 //CREATE----------------------------
+	@Override
+	@Transactional
+	public Eval createEval(Eval eval) {
+		
+		System.out.println(eval);
+		
+		if(eval.getId() != null || eval.getId() != 0) {
+			throw new ConstraintViolationException("Id is automatically generated - do not include in post request", null);
+		}
+		
+		if(eval.getTrainee() == null || eval.getTrainee().getId() == 0){
+			throw new ConstraintViolationException("Missing required field trainee.id (Integer)", null);
+		}
+		
+		if(eval.getWeek() == null || eval.getWeek() == 0){
+			throw new ConstraintViolationException("Missing required field week (Integer)", null);
+		}
+		
+		if(eval.getEvalType() == null || eval.getEvalType().getId() == 0){
+			throw new ConstraintViolationException("Missing required field evalType.id (Integer)", null);
+		}
+		
+		if(eval.getBatch() == null || eval.getBatch().getId() == 0){
+			throw new ConstraintViolationException("Missing required field batch.id (Integer)", null);
+		}
+		
+		eval.setDate(new java.sql.Date((new Date()).getTime()));
+		
+		// Set eval property of questions and qEval property of question comments
+		// This is required because these fields are not auto-generated in json mapping
+		if (eval.getQuestions() != null) {
+			eval.getQuestions().forEach((question) -> {
+				question.setEval(eval);
+				// also set qEval of question comments
+				if (question.getComments() != null) {
+					question.getComments().forEach((comment) -> {
+						comment.setQuestionEval(question);
+					});
+				}
+			});
+		}
+		
+		// Set eval property of comments so cascade works
+		if (eval.getComments() != null) {
+			eval.getComments().forEach((comment) -> {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Remove useless curly braces around statement [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1602)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove the parentheses around the "comment" parameter [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.revaturelabs.com/coding_rules#rule_key=squid%3AS1611)